ts/private/ts_project_worker.js

```diff
@@ -588,6 +591,14 @@ function emit(args, inputs) {
 
     timingStart('applyChanges')
     host.applyChanges();
+    if (failedLookups.size) {
+        timingStart('applyChanges for failedLookups');
+        for (const [input, digest] of failedLookups) {
+            host.invalidate(input, ts.FileWatcherEventKind.Changed, digest);
+        }
+        host.applyChanges();
+        timingEnd('applyChanges for failedLookups');
```

Comment:
Should start/end things like this always be within a try/finally in case something between throws?

Reply:
I guess we don't elsewhere right now though so maybe not? Either none or do them all in a separate PR...

Reply:
do we need to know how long did it take to fail? nothing in this file is supposed to fail. if it does 99% of the time it is a bug we'd have to fix.

Reply:
Depends how the timer works. Normally if one never ends then a) it never ends and essentially leaks and b) it often messes up the next one (but that depends how it's implemented).

Reply:
If the extra callback isn't a performance issue normally I'd do:

```
timer("name", () => { ... the work ... })

function timer(name, work) {
  start(name);
  try { return work() }
  finally { end(name); }
}
```

Reply:
we just delegate it to `ts.performance` api which in turn delegates to node `perf` module. https://github.com/microsoft/TypeScript/blob/main/src/compiler/performance.ts

Reply:
Interesting, those methods don't clear the data anyway so failing to call `end()` doesn't really matter 👍